### DISCUSSION TITLE
Release fails to upload deb binaries

### DISCUSSION THREAD

**Discussion by Khatkar001** [oclif/oclif#1396] [Q&A] [answered]
Hi,

Executed pack 3 times one for each platform `oclif pack deb|macos|win` to build the binaries and yields the following binaries:

```
Run ls -R ./apps/myapp-cli/dist/{deb,macos,win32}
./apps/myapp-cli/dist/deb:
myapp_1.0.0.329b9e8-1_amd64.deb
myapp_1.0.0.329b9e8-1_arm64.deb
myapp_1.0.0.329b9e8-1_armel.deb

./apps/myapp-cli/dist/macos:
myapp-v1.0.0-329b9e8-arm64.pkg
myapp-v1.0.0-329b9e8-x64.pkg

./apps/myapp-cli/dist/win32:
myapp-v1.0.0-329b9e8-x64.exe
myapp-v1.0.0-329b9e8-x86.exe
```

Both macos and win binaries take the form:
```
<APPNAME>-<VERSION>-<SHA>-<ARCHITECTURE>.<EXT>
```

However deb binaries use a different field separator `_` and include an extra `-1` after <SHA>:
```
<APPNAME>_<VERSION>_<SHA>-1_<ARCHITECTURE>.<EXT>
```

This has the knock-on effect that the `oclif upload deb` is the only platform where the upload fails as the operation can't find the binaries to upload:
```
  ›   Error: Cannot find debian artifacts
 ›   Try this: Run "oclif pack deb" before uploading
```

This seems to originate here: https://github.com/oclif/oclif/blob/a7a4abfa9a65c04782ca613defea75cf3e07889f/src/upload-util.ts#L62

```
export function debVersion(buildConfig: TarballConfig): string {
  return `${buildConfig.config.version.split('-')[0]}.${buildConfig.gitSha}-1`
  // see debian_revision: https://www.debian.org/doc/debian-policy/ch-controlfields.html
}
```

Could you confirm if it's correct to be adding the extra `-1` for the debian builds?

If it is correct, why does the `upload` method not detect the binary filename?

**Comment by mdonnalley** [comment, marked as answer]
@Khatkar001 The `upload deb` command is exiting early because there's no `Release` file under `dist/deb`. You can see that check [here](https://github.com/oclif/oclif/blob/main/src/commands/upload/deb.ts#L27)

The `Release` file should have been created during `pack deb` [here](https://github.com/oclif/oclif/blob/main/src/commands/pack/deb.ts#L143). Not sure why that wouldn't be working for you though. Did you see any errors while running `oclif pack deb`?

**Reply by Khatkar001** [reply]
Beautiful!

Yep the GitHub action was only uploading *.deb artifacts, uploading them all has resolved it.

Thank you!
